pkg/kubectl/cmd/expose.go

```diff
@@ -112,6 +115,9 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		return fmt.Errorf("multiple resources provided: %v", args)
 	}
 	info := infos[0]
+	if info.Mapping.Kind != "ReplicationController" && info.Mapping.Kind != "Service" && info.Mapping.Kind != "Pod" {
```

Comment:
@feihujiang create a `CanBeExposed func(kind string) bool` method in the factory and handle this check in [there](https://github.com/kubernetes/kubernetes/blob/196f58b9cb25a2222c7f9aacd624737910b03acb/pkg/kubectl/cmd/util/factory.go#L50).